File: Backend/proyecto/src/main/java/com/admindb/proyecto/modelo/permisos/PermisoTablaId.java
package com.admindb.proyecto.modelo.permisos;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import lombok.Getter;
import lombok.Setter;
import lombok.RequiredArgsConstructor;
import lombok.EqualsAndHashCode;



@Getter
@Setter
@Embeddable
@EqualsAndHashCode
@RequiredArgsConstructor
public class PermisoTablaId implements Serializable{
    
    @Column(name = "table_name")
    private String nombreTabla;

    @Column(name = "privilege")
    private String privilegio;
}

File: Frontend/frontend-adminbd/src/app/common/permiso-tabla-id.ts
export class PermisoTablaId {

    nombreTabla: string;

    privilegio: string;

    otorgado: string;

    constructor(nombreTabla: string, privilegio: string, otorgado: string){
        this.nombreTabla = nombreTabla;
        this.privilegio = privilegio;
        this.otorgado = otorgado;
    }

}
